src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -14,18 +14,44 @@ interface DropdownMenuItemProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   className?: string;
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  // To select a certain option in the menu without having to use document.querySelector
+  customRef?: React.RefObject<HTMLDivElement>;
+  onActivate?: (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) => void;
 }
 
 export const DropdownMenuDivider: React.FC = () => <div className="tk-dropdown-menu-divider"></div>
 
-export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, ...rest}: DropdownMenuItemProps) => {
+export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, customRef, onActivate, ...rest}: DropdownMenuItemProps) => {
   const classes = classNames(
     'tk-dropdown-menu__item',
     className,
   )
 
+  const focusNextOption = (current: HTMLDivElement, direction: number) => {
+    const options = (Array.from(current.parentElement.getElementsByClassName('tk-dropdown-menu__item'))) as HTMLDivElement[];
+    let currentOptionPosition = options.indexOf(current);
+    currentOptionPosition = currentOptionPosition > -1 ? currentOptionPosition : 0;
+    const nextElementPosition = (currentOptionPosition + options.length + direction) % options.length;
+
+    options[nextElementPosition].focus();
+  }
+
+  const onKeyDownHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    e.stopPropagation();
+    switch (e.key) {
```

Comment:
We have a enum for keys, that you can reuse
https://github.com/SymphonyPlatformSolutions/symphony-bdk-ui-toolkit-components/blob/master/src/components/common/keyUtils.tsx

Reply:
also a `cancelEvent` util method :)

Reply:
`cancelEvent` sounds a bit confusing, sounds like you are gonna cancel the fired event  😅.
Personally I think it's clearer to just have `e.stopPropagation`, plus it's not really necessary here to have `e.preventDefault()`here.

Reply:
ok!